tests/west/project/test_project.py

```diff
@@ -1,168 +1,352 @@
-import argparse
 import os
+from os.path import dirname
 import shlex
 import shutil
 import subprocess
 import sys
+import textwrap
+from unittest.mock import patch
 
 import pytest
 
-from west import config
-from west.commands import project
-import west._bootstrap.main as bootstrap
-
-# Where the projects are cloned to
-NET_TOOLS_PATH = 'net-tools'
-KCONFIGLIB_PATH = 'sub/Kconfiglib'
+import west._bootstrap.main
 
 GIT = shutil.which('git')
+# Assumes this file is west/tests/west/project/test_project.py, returns
+# path to toplevel 'west'
+THIS_WEST = os.path.abspath(dirname(dirname(dirname(dirname(__file__)))))
```

Comment:
Not the prettiest but I guess in a test it doesn't matter.

Reply:
Yeah, it's pretty hideous.